src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java

```diff
@@ -0,0 +1,85 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.exam.StudentExam;
+import de.tum.in.www1.artemis.repository.StudentExamRepository;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * Service Implementation for managing StudentExam.
+ */
+@Service
+public class StudentExamService {
+
+    private final Logger log = LoggerFactory.getLogger(StudentExamService.class);
+
+    private final StudentExamRepository studentExamRepository;
+
+    private final ExamAccessService examAccessService;
+
+    public StudentExamService(StudentExamRepository studentExamRepository, ExamAccessService examAccessService) {
+        this.studentExamRepository = studentExamRepository;
+        this.examAccessService = examAccessService;
+    }
+
+    /**
+     * Get one student exam by id.
+     *
+     * @param studentExamId the id of the student exam
+     * @return the entity
+     */
+    @NotNull
+    public StudentExam findOne(Long studentExamId) {
+        log.debug("Request to get student exam : {}", studentExamId);
+        return studentExamRepository.findById(studentExamId).orElseThrow(() -> new EntityNotFoundException("Student exam with id \"" + studentExamId + "\" does not exist"));
+    }
+
+    /**
+     * Get all student exams for the given exam.
+     *
+     * @param examId the id of the exam
+     * @return the list of all student exams
+     */
+    public List<StudentExam> findAllByExamId(Long examId) {
+        log.debug("REST request to get all student exams for Exam : {}", examId);
+        return studentExamRepository.findByExamId(examId);
+    }
+
+    /**
+     * Checks if the current user is allowed to manage exams of the given course, that the exam exists,
+     * that the exam belongs to the given course and the student exam belongs to the given exam.
+     *
+     * @param courseId      The id of the course
+     * @param examId        The id of the exam
+     * @param studentExamId The if of the student exam
+     * @param <X>           The type of the return type of the requesting route so that the
+     *      *               response can be returned there
+     * @return an Optional with a typed ResponseEntity. If it is empty all checks passed
+     */
+    public <X> Optional<ResponseEntity<X>> checkCourseAndExamAndStudentExamAccess(Long courseId, Long examId, Long studentExamId) {
+        Optional<ResponseEntity<X>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccess(courseId, examId);
+        if (courseAndExamAccessFailure.isPresent()) {
+            return courseAndExamAccessFailure;
+        }
+        Optional<StudentExam> studentExam = studentExamRepository.findById(studentExamId);
+        if (studentExam.isEmpty()) {
+            return Optional.of(notFound());
+        }
+        if (!studentExam.get().getExam().getId().equals(examId)) {
+            return Optional.of(forbidden());
```

Comment:
I don't think forbidden() is a good choice here. The user is allowed however the student exam doesn't belong to the exam.
You could either agrue with a "notFound()" here since you can't finde an exercise belonging to the course, maybe a HTTP 409 would be a good choice here?

Reply:
I've added `conflict()` to `ResponseUtil` and applied it here.